compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1425,7 +1425,8 @@ impl AcirContext {
         index: &AcirVar,
     ) -> Result<AcirVar, InternalError> {
         // Fetch the witness corresponding to the index
-        let index_witness = self.var_to_witness(*index)?;
+        let index_var = self.get_or_create_witness_var(*index)?;
+        let index_witness = self.var_to_witness(index_var)?;
```

Comment:
If we're going to be using this pattern commonly, then it makes sense to create a new method on self which calls both of these methods

Reply:
I think we can handle this in a followup. There's a couple of places where we need different handling and naming 3 different methods which all do very similar things is something that can get bikesheddy.